src/promisifyRequest.ts

```diff
@@ -0,0 +1,69 @@
+import { RequestCallback } from './types';
+
+/**
+ * Wraps a method for returning the promise
+ * @param fn a method for wrapping
+ * @param returnObject should return a object with field promise and field with return value of method (Default: `false`)
+ * @returns wrapped method
+ */
+export function promisifyRequest<
+  Result,
+  MethodReturn,
+  Params extends {} | undefined,
+  ReturnObject extends true | false | undefined = undefined,
+>(
+  fn: (params: RequestParams<Result> & Params) => MethodReturn,
+  returnObject?: ReturnObject
```

Comment:
`hasReturnObject` or `hasReturnValue`? And why not simply Boolean?

Reply:
I like `hasReturnObject`

> And why not simply Boolean?

For more detailing typings. Return value of the method depends by type of this argument.

Reply:
Now using boolean. With it type work too